src/ConnectionFromMongoCursor.js

```diff
@@ -170,7 +170,10 @@ const connectionFromMongoCursor = async ({
   cursor.skip(skip);
   cursor.limit(limit);
 
-  const slice = await cursor.exec();
+  //avoid large object retrieval from database
+  const clonedCursor = cursor.find().merge(cursor).select({ _id: 1 })
```

Comment:
Looking at the code http://mongoosejs.com/docs/api.html#query_Query-merge this method is nothing more then an Object.assign